contrib/src/main/java/org/apache/apex/malhar/contrib/kudu/AbstractKuduOutputOperator.java

```diff
@@ -0,0 +1,666 @@
+  {
+    super.setup(context);
+    windowDataManager = getWindowDataManager();
+    if ( windowDataManager == null) {
+      windowDataManager = new FSWindowDataManager();
+    }
+    windowDataManager.setup(context);
+  }
+
+  @Override
+  public void teardown()
+  {
+    super.teardown();
+    windowDataManager.teardown();
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    currentWindowId = windowId;
+    if ( currentWindowId != Stateless.WINDOW_ID) {
+      if (currentWindowId > reconcilingWindowId) {
+        isInSafeMode = false;
+        isInReconcilingMode = false;
+      }
+      if (currentWindowId == reconcilingWindowId) {
+        isInReconcilingMode = true;
+        isInSafeMode = false;
+      }
+      if (currentWindowId < reconcilingWindowId) {
+        isInReconcilingMode = false;
+        isInSafeMode = true;
+      }
+    }
+    numDeletedInThisWindow = 0;
+    numInsertsInThisWindow = 0;
+    numUpdatesInThisWindow = 0;
+    numUpsertsInThisWindow = 0;
+  }
+
+  @Override
+  public void endWindow()
+  {
+    super.endWindow();
+    try {
+      kuduSession.flush();
+    } catch (KuduException e) {
+      LOG.error("Could not flush kudu session on an end window boundary " + e.getMessage(), e);
+      throw new RuntimeException(e.getMessage());
+    }
+    try {
+      windowDataManager.save(currentWindowId,currentWindowId);
+    } catch (IOException e) {
+      LOG.error("Error while persisting the current window state " + currentWindowId + " because " + e.getMessage());
+      throw new RuntimeException(e.getMessage());
+    }
+    opsErrorsInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.OPS_ERRORS) -
+      totalOpsErrors;
+    bytesWrittenInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.BYTES_WRITTEN) -
+      totalBytesWritten;
+    rpcErrorsInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.RPC_ERRORS) -
+      totalRpcErrors;
+    writeOpsInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.WRITE_OPS) -
+      totalWriteOps;
+    writeRPCsInThisWindow = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.WRITE_RPCS) -
+      totalWriteOps;
+    totalOpsErrors = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.OPS_ERRORS);
+    totalBytesWritten = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.BYTES_WRITTEN);
+    totalRpcErrors = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.RPC_ERRORS);
+    totalWriteOps = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.WRITE_OPS);
+    totalWriteRPCs = kuduClientHandle.getStatistics().getClientStatistic(Statistics.Statistic.WRITE_RPCS);
+  }
+
+  private void buildGettersForPojoPayload()
+  {
+    Class payloadClass = getTuplePayloadClass();
```

Comment:
Resolved this implementing a Default implementation for the Abstract Kudu output operator. This default implementation would accept a properties file ( along with a no arg constructor to use a default property file name ). The property file would help to specify the above value as well as a few others. The java docs were updated to reflect this accordingly.